azurerm/internal/services/network/data_source_peer_asn.go

```diff
@@ -0,0 +1,51 @@
+package network
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/network/validate"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func dataSourceArmPeerAsn() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceArmPeerAsnRead,
+
+		Timeouts: &schema.ResourceTimeout{
+			Read: schema.DefaultTimeout(5 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
```

Comment:
Are there many instances where the user will need to look up an ASN via name to get the id?

Reply:
User will need that id in order to create peering (another resource).